File: 0768-partition-labels/0768-partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        lastidx = [-1]*26

        for i,c in enumerate(s): 
            lastidx[ord(c) - ord('a')] = i

        res = []
        sz, end = 0, 0
        for i, c in enumerate(s):
            sz += 1
            end = max(end, lastidx[ord(c) - ord('a')])
            
            if i == end:
                res.append(sz)
                sz = 0

        return res

File: 0768-partition-labels/0768-partition-labels.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        vector<int> lastidx(26, -1); 
        
        for (int i = 0; i < s.size(); i++) {
            lastidx[s[i] - 'a'] = i;
        }

        vector<int> res;
        int sz = 0, end = 0;
        
        for (int i = 0; i < s.size(); i++) {
            sz++;
            end = max(end, lastidx[s[i] - 'a']); 
            
            if (i == end) {
                res.push_back(sz);
                sz = 0;
            }
        }

        return res;        
    }
};